lib/ts/recipe/thirdparty/recipeImplementation.ts

```diff
@@ -192,29 +159,37 @@ export default function getRecipeImplementation(recipeId: string, appInfo: Norma
                   status: "NO_EMAIL_GIVEN_BY_PROVIDER";
                   fetchResponse: Response;
               }
-            | {
-                  status: "FIELD_ERROR";
-                  error: string;
-                  fetchResponse: Response;
-              }
         > {
-            const stateFromStorage = this.getOAuthState({
+            const stateFromStorage = this.getStateAndOtherInfoFromStorage<{}>({
                 userContext: input.userContext,
                 config: input.config,
-            }).state;
-
-            const code = getQueryParams("code");
+            });
 
             const stateFromQueryParams = getQueryParams("state");
 
             if (
-                getQueryParams("error") !== null ||
                 stateFromStorage === undefined ||
```

Comment:
You should remove this undefined check as it should be done inside `verifyStateFromOAuthProvider`